res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1336 @@
+    "[Channel1]": 0,
+    "[Channel2]": 0
+  };
+  this.previous_leftencoder = {
+    "[Channel1]": 0,
+    "[Channel2]": 0
+  };
+  this.wheelTouchInertiaTimer = {
+    "[Channel1]": 0,
+    "[Channel2]": 0
+  };
+  
+  this.top_encoder_pressed = {
+    "[Channel1]": false,
+    "[Channel2]": false
+  };
+  this.left_encoder_pressed = {
+    "[Channel1]": false,
+    "[Channel2]": false
+  };
+  this.shift_pressed = {
+    "[Channel1]": false,
+    "[Channel2]": false
+  };
+                                   
+  this.pad_modes = {
+    "hotcue": 0,
+    "intro_outro": 1,
+    "sampler": 2
+  };
+  this.current_pad_mode = {
+    "[Channel1]": this.pad_modes.hotcue,
+    "[Channel2]": this.pad_modes.hotcue
+  };
+  this.pad_connections = {
+    "[Channel1]": [],
+    "[Channel2]": []
+  };
+
+  this.last_tick_val = [0, 0];
+  this.last_tick_time = [0.0, 0.0];
+  this.sync_enabled_time = {};
+  
+  this.longPressTimeout = 275; // milliseconds
+  
+  this.effectButtonLongPressTimer = {
+    "[EffectRack1_EffectUnit1]": [0, 0, 0, 0],
+    "[EffectRack1_EffectUnit2]": [0, 0, 0, 0]
+  };
+  this.effectButtonIsLongPressed = {
+    "[EffectRack1_EffectUnit1]": [false, false, false, false],
+    "[EffectRack1_EffectUnit2]": [false, false, false, false]
+  };
+  this.effectFocusLongPressTimer = {
+    "[EffectRack1_EffectUnit1]": 0,
+    "[EffectRack1_EffectUnit2]": 0
+  };
+  this.effectFocusChooseModeActive = {
+    "[EffectRack1_EffectUnit1]": false,
+    "[EffectRack1_EffectUnit2]": false
+  };
+  this.effectFocusButtonPressedWhenParametersHidden = {
+    "[EffectRack1_EffectUnit1]": false,
+    "[EffectRack1_EffectUnit2]": false
+  };
+  this.previouslyFocusedEffect = {
+    "[EffectRack1_EffectUnit1]": null,
+    "[EffectRack1_EffectUnit2]": null
+  };
+  this.effectButtonLEDconnections = {
+    "[EffectRack1_EffectUnit1]": [],
+    "[EffectRack1_EffectUnit2]": []
+  };
+}
+
+TraktorS2MK2.registerInputPackets = function() {
+  var MessageShort = new HIDPacket("shortmessage", 0x01, this.shortMessageCallback);
+  var MessageLong = new HIDPacket("longmessage", 0x02, this.longMessageCallback);
+
+  // Values in the short message are all buttons, except the jog wheels.
+  // An exclamation point indicates a specially-handled function.  Everything else is a standard
+  // Mixxx control object name.
+  // "[Channel1]" and "[Channel2]" refer to the left deck or right deck, and may be Channel1 or 3 depending
+  // on the deck switch state.  These are keywords in the HID library.
+  
+  MessageShort.addControl("[Channel1]", "!top_encoder_press", 0x0D, "B", 0x40, false, this.topEncoderPress);  
+  MessageShort.addControl("[Channel1]", "!shift", 0x0B, "B", 0x08, false, this.shift);
+  MessageShort.addControl("[Channel1]", "!sync_enabled", 0x0B, "B", 0x04, false, this.syncEnabled);
+  MessageShort.addControl("[Channel1]", "!cue_default", 0x0B, "B", 0x02, false, this.cue);
+  MessageShort.addControl("[Channel1]", "!play", 0x0B, "B", 0x01, false, this.play);
```

Comment:
Please make the handlers playButton or handlePlay for consistency.  Same with other handlers.